tests/tests/format/urls.php

```diff
@@ -80,6 +81,9 @@ function list_of_valid_URLs() {
             array( '/absolute' ),
             array( '/Absolute/Path/' ),
             array( '/absolute/path/?omg#also' ),
+            array( 'http://académie-française.fr' ),
+            array( 'http://www.طارق.net/طارق?hello=%2B' ),
+            array( 'http://%d8%b7%d8%a7%d8%b1%d9%82.net/' ), // this is طارق.net, encoded. I _think_ it qualifies as valid
```

Comment:
Well, not sure at all. IDN should keep their official encoding system, here: `http://xn--mgbuq0c.net/`
Best option is to only store `xn--` version of domains.

Reply:
I added this test because during my testing (shorten normally in admin area, use bookmarklet, use `http://sho.rt/` prefix) I ended up with this. Browsers are happy with it...